substrate/frame/scheduler/src/lib.rs

```diff
@@ -1192,6 +1320,52 @@ impl<T: Config> Pallet<T> {
 		let _ = weight.try_consume(call_weight);
 		Ok(result)
 	}
+
+	/// Check if a task has a retry configutation in place and, if so, try to reschedule it.
+	///
+	/// If the task was successfully rescheduled for later, this function will return `None`.
+	/// Otherwise, if the task was not rescheduled, either because there was no retry configuration
+	/// in place, there were no more retry attempts left, or the agenda was full, this function
+	/// will return the task so it can be handled somewhere else.
+	fn check_retry(
+		now: BlockNumberFor<T>,
+		when: BlockNumberFor<T>,
+		agenda_index: u32,
+		task: ScheduledOf<T>,
+	) -> Option<ScheduledOf<T>> {
+		// Check if we have a retry configuration set.
+		match Retries::<T>::take((when, agenda_index)) {
+			Some(RetryConfig { total_retries, mut remaining, period }) => {
+				remaining = match remaining.checked_sub(1) {
+					Some(n) => n,
+					None => return Some(task),
+				};
+				let wake = now.saturating_add(period);
+				match Self::place_task(wake, task) {
+					Ok(address) => {
+						// Reinsert the retry config to the new address of the task after it was
+						// placed.
+						Retries::<T>::insert(
+							address,
+							RetryConfig { total_retries, remaining, period },
+						);
+						None
+					},
+					Err((_, task)) => {
+						// TODO: Leave task in storage somewhere for it to be
+						// rescheduled manually.
+						T::Preimages::drop(&task.call);
+						Self::deposit_event(Event::RetryFailed {
+							task: (when, agenda_index),
+							id: task.maybe_id,
+						});
+						Some(task)
+					},
+				}
+			},
+			None => Some(task),
+		}
+	}
 }
 
 impl<T: Config> schedule::v2::Anon<BlockNumberFor<T>, <T as Config>::RuntimeCall, T::PalletsOrigin>
```

Comment:
I think it would be easier to just include an optional `RetryConfig` as part of the `schedule` and `schedule_named` call instead of having another call, or?

Reply:
I think this is a niche mechanic and I wouldn't burden the `schedule` and `schedule_named` extrinsics with this, they already hide enough complexity as it stands. It's simpler to test as well. Also, doing it separately like this also has the nice bonus of being backwards compatible. Do you feel strongly about this?

Reply:
Hm okay. Yea its probably not used that often.